src/edge_utils.c

```diff
@@ -3000,6 +3035,14 @@ void edge_term (n2n_edge_t * eee) {
     clear_peer_list(&eee->known_peers);
     clear_peer_list(&eee->conf.supernodes);
 
+    if (eee->conf.allow_routing) {
+        struct host_info *host, *host_tmp;
```

Comment:
Not sure if this will trigger compiler warning, maybe define at top of function (same for other definitions)?

Reply:
As the variables are used just by this codeblock，It is safer to be defined as local variables inside the codeblock.

Reply:
Yea, but it does not match the style of coding we have been trying to keep up so far... ;)

Reply:
We will leave it for now as it also allows for easier feature-define...